app/src/main/java/io/gnosis/safe/ui/adapter/BaseAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package io.gnosis.safe.ui.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+
+class BaseAdapter<VH, T>(
+    private val factory: BaseFactory<VH>
+) : RecyclerView.Adapter<VH>() where VH : BaseViewHolder<T> {
+
+    private val items = mutableListOf<T>()
+
+    @Deprecated("Unsafe")
+    fun setItemsUnsafe(items: List<T>) {
```

Comment:
Do we plan on using the DiffUtils?

Reply:
Absolutely. We already have one adapter that makes use of it afaik. This is a bit of a naive implementation. I just need to verify what's the scope of the issue defined xD